File: Backend-Java/src/main/java/org/example/Infrastructure/dto/categoryDTO/CategoryEditDTO.java
package org.example.Infrastructure.dto.categoryDTO;

import lombok.Data;
import org.springframework.web.multipart.MultipartFile;
@Data
public class CategoryEditDTO {
    private int id;
    private String name;
    private MultipartFile imageUpload;
    private String description;
    private Integer parentId;
}

File: Backend-C#/Infrastructure.MyShop/Models/DTO/CategoryDTO/CategoryEditDTO.cs
using Microsoft.AspNetCore.Http;

namespace Infrastructure.MyShop.Models.DTO.CategoryDTO
{
    public class CategoryEditDTO
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public IFormFile? ImageUpload { get; set; } // IFormFILE || null
        public string Description { get; set; }
        public int? ParentId { get; set; } // ParentId || null
    }
}
